src/utils/account-with-lockup.js

```diff
@@ -34,22 +37,36 @@ async function signAndSendTransaction(receiverId, actions) {
         console.warn('Not enough balance on main account, checking lockup account', lockupAccountId);
 
         if (!(await this.wrappedAccount.viewFunction(lockupAccountId, 'are_transfers_enabled'))) {
-            await this.wrappedAccount.functionCall(lockupAccountId, 'check_transfers_vote', {}, new BN(75000000000000))
-        }
-
-        const liquidBalance = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_liquid_owners_balance'))
-        if (!liquidBalance.gt(missingAmount)) {
-            throw new WalletError('Not enough tokens.', 'sendMoney.amountStatusId.notEnoughTokens')
+            await this.wrappedAccount.functionCall(lockupAccountId, 'check_transfers_vote', {}, BASE_GAS.mul(new BN(3)))
         }
 
-        // TODO: better gas calculation (use base amount multiples)
-        await this.wrappedAccount.functionCall(lockupAccountId, 'transfer', {
-            // NOTE: Move all the liquid tokens to minimize transactions in the long run
-            amount: liquidBalance.toString(),
-            receiver_id: this.wrappedAccount.accountId
-        }, new BN(50000000000000))
+        const knownDeposited = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_known_deposited_balance'))
+        const lockedBalance = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_locked_amount'))
+        if (knownDeposited.eq(new BN(0)) && lockedBalance.eq(new BN(0))) {
```

Comment:
I'd also recommend to make sure there is no pool selected or to refresh staking balance first, if the pool is selected. It's unlikely that there are still some balance on the pool, but when you delete a lockup contract, you'll not be able to recover this balance from the staking pool anymore.

The logic only requires that the `get_locked_amount` is `0` and it doesn't care about the staking amount.

Reply:
@evgenykuzyakov shouldn't `get_known_deposited_balance` be strictly greater than zero though if anything is deposited to pool?

> but when you delete a lockup contract, you'll not be able to recover this balance from the staking pool anymore.

I guess technically you still can if same-named lockup contract is created, correct?

Reply:
Is it possible to query balance deposited to pool without making transaction (e.g. by making view call to the pool)?

Reply:
> shouldn't get_known_deposited_balance be strictly greater than zero though if anything is deposited to pool?

No, because you may receive rewards on the pool, but without refreshing the pool balance withdraw the original staked amount. This reward will not be reported to the lockup contract.

Reply:
@evgenykuzyakov so if I call withdraw, etc methods balance is not refreshed? When is balance refreshed? Should I really rely on anything in lockup contract at all or better call pool contract?

> Is it possible to query balance deposited to pool without making transaction (e.g. by making view call to the pool)?

this isn't answered

Reply:
> Is it possible to query balance deposited to pool without making transaction (e.g. by making view call to the pool)?

Yes and no. You can view balance on the staking pool, by calling `get_account_total_balance(lockup_account_id)`, but it doesn't include rewards if the epoch switched and no one called `ping` or interacted with it.

Reply:
@evgenykuzyakov I don't care about rewards though. I just need to check whether there is anything deposited. Will querying `get_account_total_balance` be reliable enough for that?

Reply:
> @evgenykuzyakov I don't care about rewards though. I just need to check whether there is anything deposited. Will querying get_account_total_balance be reliable enough for that?

As Evgeny pointed out it is not sufficient to call `get_account_total_balance` because the staking pool does lazy updates and it might not have been updated yet to reflect your true balance. Therefore it is safer to delete lockup after you refresh balance or when there is no pool selected, as Evgeny suggested.

Reply:
Basically to be safe to delete lockup, you need to do the following:
- ping pool (it sucks that we have to do this)
- view `<staking_pool>.get_account_total_balance(<lockup_account_id)` to check if the lockup still has balance
- if anything meaningful left that the user might be upset (e.g. more than 1 NEAR), then `unstake_all` then wait and `withdraw_all_from_staking_pool` and `unselect_staking_pool`. Or ask for confirmation from the user.

If we want to be fast. Check `<lockup_account_id>.get_known_deposited_balance() == 0` and `get_locked_amount() == 0`

Reply:
@bowenwang1996  @evgenykuzyakov I don't understand why it is necessary to ping pool. Shouldn't every change operation using a pool already ping it by default? If so – I cannot understand when exactly you won't have 0 balance being accurate.

Reply:
Note that I don't care about having accurate number if it's not 0